File: c#/src/v1/models/MessagePriority.cs
using System.Text.Json.Serialization;

namespace Comms.Models;

public enum MessagePriority
{
    [JsonPropertyName("0")]
    Highest,
    [JsonPropertyName("1")]
    High,
    [JsonPropertyName("2")]
    Medium,
    [JsonPropertyName("3")]
    Low,
    [JsonPropertyName("4")]
    Lowest,
}

File: kotlin/src/main/kotlin/v1/models/MessagePriority.kt
package v1.models

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

enum class MessagePriority(private val value: String) {
    HIGHEST("0"),
    HIGH("1"),
    MEDIUM("2"),
    LOW("3"),
    LOWEST("4");

    @JsonValue
    override fun toString(): String {
        return value
    }

    companion object {
        @JsonCreator
        fun fromValue(text: String?): MessagePriority {
            for (priority in MessagePriority.entries) {
                if (priority.value == text) {
                    return priority
                }
            }
            throw IllegalArgumentException("Unknown priority value: $text")
        }
    }
}
